datastores/run_data.ts

```diff
@@ -21,4 +24,44 @@ const RunningDatastore = DefineDatastore({
   },
 });
 
+/**
+ * Returns the complete collection from the datastore for an expression
+ *
+ * @param client the client to interact with the Slack API
+ * @param expressions optional filters and attributes to refine a query
+ *
+ * @returns ok if the query completed successfully
+ * @returns items a list of responses from the datastore
+ * @returns error the description of any server error
+ */
+export async function queryRunningDatastore(
+  client: SlackAPIClient,
+  expressions?: object,
+): Promise<{
+  ok: boolean;
+  items: DatastoreItem<typeof RunningDatastore.definition>[];
+  error?: string;
+}> {
+  const items: DatastoreItem<typeof RunningDatastore.definition>[] = [];
+  let cursor = undefined;
+
+  do {
+    const runs: DatastoreQueryResponse<typeof RunningDatastore.definition> =
+      await client.apps.datastore.query<typeof RunningDatastore.definition>({
+        datastore: RUN_DATASTORE,
+        cursor,
+        ...expressions,
+      });
+
+    if (!runs.ok) {
+      return { ok: false, items, error: runs.error };
+    }
+
+    cursor = runs.response_metadata?.next_cursor;
+    items.push(...runs.items);
+  } while (cursor);
```

Comment:
`do`/`while` w/ cursors in Dynamo is exactly how I structure my DDB pagination calls 👍